sycl/test-e2e/Basic/cospi.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <iomanip>
+#include <iostream>
+#include <sycl.hpp>
+
+template <typename T> T get_ulp_std(T x) {
+  const T inf = std::numeric_limits<T>::infinity();
+  const T negative = std::fabs(std::nextafter(x, -inf) - x);
+  const T positive = std::fabs(std::nextafter(x, inf) - x);
+  return std::fmin(negative, positive);
+}
+
+int main() {
+  const double value = 0.4863334355; // some random value
+  const double reference =
+      0.042921588887841428949787569990803604014217853546142578125; // calculated
+                                                                   // with
+                                                                   // oclmath
+  double res;
+
+  {
+    sycl::queue queue;
+    sycl::buffer<double, 1> cospi_buf(&res, 1);
+    queue
+        .submit([&](sycl::handler &cgh) {
+          auto cospi_acc = cospi_buf.get_access(cgh);
+          cgh.single_task([=]() { cospi_acc[0] = sycl::cospi(value); });
```

Comment:
The test for device `sycl::cospi` is here: https://github.com/intel/llvm/blob/6ba140044371a85af97e051fb19b53d948721992/sycl/test-e2e/DeviceLib/built-ins/scalar_math.cpp#L168
Suggest not to duplicate it and remove from this PR.